src/TConsole.cpp

```diff
@@ -2949,3 +2950,41 @@ void TConsole::setProfileName(const QString& newName)
         pC->setProfileName(newName);
     }
 }
+
+
+void TConsole::dragEnterEvent(QDragEnterEvent* e)
+{
+    if (e->mimeData()->hasUrls()) {
+        e->acceptProposedAction();
+    }
+}
+
+//https://amin-ahmadi.com/2016/01/04/qt-drag-drop-files-images/
+void TConsole::dropEvent(QDropEvent* e)
+{
+    QStringList accepted_types;
+    accepted_types << "jpeg"
+                   << "jpg"
+                   << "png";
+    foreach(const QUrl& url, e->mimeData()->urls()) {
+        QString fname = url.toLocalFile();
+        QFileInfo info(fname);
+        if (info.exists()) {
+            if (accepted_types.contains(info.suffix().trimmed(), Qt::CaseInsensitive)) {
```

Comment:
Good point to test the extension case insensitively - some OSes file-systems (macOs) are not fully case sensitive.